external/avro/src/test/scala/org/apache/spark/sql/avro/AvroCatalystDataConversionSuite.scala

```diff
@@ -127,6 +127,25 @@ class AvroCatalystDataConversionSuite extends SparkFunSuite
     }
   }
 
+  test(s"array of nested schema with seed") {
```

Comment:
I tried the test case with `GenericData.Array` and I can't reproduce the error you mentioned.

Reply:
@gengliangwang you are right.  This is an extra case which covers an extra scenario, but is was not triggering this issue.  (I thought it did but it did not.)

I added an extra test to trigger the issue.  I am wondering what you think of it.
Hope it does not feel too artificial.  It tries to show that a GenericData can contain different list implementation types.

We hit this issue where we map an `RDD[T]` to a `DataFrame` using the `AvroDeserializer`.  Where `T` is a class generate by the avro code generator.  So it extends `SpecifiData` which extends `GenericData`.  So all the code of the `AvroDeserializer` works, except of this cast, which casts to a type which is too high in the type hierarchy.

Hope it is more clear now.

Thanks for your support.